performance_search_max_throughput_test.py

```diff
@@ -0,0 +1,200 @@
+
+                threads = (best_result["total_threads"] // (len(self.loaders.nodes) * stress_num)) - threads_step
+                threads = 10 * round(threads / 10)
+                decrease_loaders_num = False
+
+        self.loaders.nodes = original_nodes_list[:]
+
+        self.log.info("Found best configuration with results %s",  best_result)
+
+        self.log.info("Write data to files")
+        filename = f"{self.logdir}/all_stats_result.json"
+        with open(filename, "w", encoding="utf-8") as fp_json:
+            json.dump(all_results, fp_json, indent=4)
+        filename = f"{self.logdir}/best_stat_result.json"
+        with open(filename, "w", encoding="utf-8") as fp_json:
+            json.dump(best_result, fp_json, indent=4)
+
+        raw_results = {
+            "best_stat": best_result,
+            "all_stats": all_results
+        }
+
+        # content = json.dumps(raw_results, indent=4, separators=(", ", ": "))
+
+        # self.log.info('Send email to {}'.format(self.params.get("email_recipients")))
+        # email = Email()
+        # email.send("Search best performance results ", content, html=False, recipients=["alex.bykov@scylladb.com"])
+
+        setup_details = {
+            "scylla_version": self.db_cluster.nodes[0].scylla_version_detailed,
+            "start_time": format_timestamp(self.start_time),
+            "instance_type_db": self.params.get("instance_type_db"),
+            "instance_type_loader": self.params.get("instance_type_loader")
+        }
+
+        analyzer = SearchBestThroughputConfigPerformanceAnalyzer(es_index=self._test_index,
+                                                                 es_doc_type=self._es_doc_type,
+                                                                 email_recipients=self.params.get("email_recipients"))
+        analyzer.check_regression(test_name, setup_details=setup_details, test_results=raw_results)
+
+    def _get_stress_parameters(self) -> dict[Any, Any]:
+        return {
+            "stress_num":  int(self.params.get("n_stress_process")),
+            "stress_num_step": int(self.params.get("stress_process_step")),
+            "num_loaders_step": int(self.params.get("num_loaders_step")),
+            "start_threads":  int(self.params.get("stress_threads_start_num")),
+            "threads_step":  int(self.params.get("num_threads_step")),
+            "max_deviation": float(self.params.get("max_deviation")),
+            "stress_step_duration":  self.params.get("stress_step_duration"),
+        }
+
+    @staticmethod
+    def _calculate_average_stats(results):
+        status = defaultdict(float).fromkeys(results[0].keys(), 0.0)
+        for result in results:
+            for key in status:
+                try:
+                    status[key] += float(result.get(key, 0.0))
+                except ValueError:
+                    continue
+
+        for key in status:
+            if key == "op rate":
```

Comment:
try to add comments to lines like this: e.g. here I think it was meant that `op rate` we sum and others we average, a comment stating that would speed up understanding.

Reply:
fixed. is it clear now?

Reply:
now code is clearer: for 'op rate' we do sum for the others average.
Can't we use `round(mean(status[key]), 2)` for calculating average?

Reply:
it require import statistics lib. I don't see the need for that for such simple operation